pvlib/tools.py

```diff
@@ -425,3 +425,112 @@ def _golden_sect_DataFrame(params, VL, VH, func):
             raise Exception("EXCEPTION:iterations exceeded maximum (50)")
 
     return func(df, 'V1'), df['V1']
+
+
+def latitude_to_geocentric(phi):
+    """
+    Converts a geodetic (common) latitude to a geocentric latitude.
+    Latitude must be given in radians.
+    [1] https://www.oc.nps.edu/oc2902w/coord/coordcvt.pdf
+    """
+    # Equatorial Radius of the Earth (ellipsoid model) in meters
+    a = 6378137.0
+    # Polar Radius of the Earth (ellipsoid model) in meters
+    b = 6356752.0
+    return np.arctan(b**2/a**2*np.tan(phi))
+
+
+def latitude_to_geodetic(phi):
+    """
+    Converts a geocentric latitude to a geodetic (common) latitude.
+    Latitude must be given in radians.
+    [1] https://www.oc.nps.edu/oc2902w/coord/coordcvt.pdf
+    """
+    # Equatorial Radius of the Earth (ellipsoid model) in meters
+    a = 6378137.0
+    # Polar Radius of the Earth (ellipsoid model) in meters
+    b = 6356752.0
+    return np.arctan(a**2/b**2*np.tan(phi))
+
+
+def lle_to_xyz(point):
+    """
+    Converts a (lat, lon, elev) tuple into a (x, y, z) tuple.
+    The center of the earth is the origin in the xyz-space.
+    The input latitude is assumed to be a common latitude (geodetic).
+    """
+    lat = np.atleast_1d(point.T[0])
+    lon = np.atleast_1d(point.T[1])
+    elev = np.atleast_1d(point.T[2])
+
+    # Equatorial Radius of the Earth (ellipsoid model) in meters
+    a = 6378137.0
+    # Polar Radius of the Earth (ellipsoid model) in meters
+    b = 6356752.0
+
+    # convert to radians
+    phi = np.radians(lat)
+    theta = np.radians(lon)
+
+    # compute radius of earth at each point
+    r = (a**2 * np.cos(phi))**2 + (b**2 * np.sin(phi))**2
+    r = r / (a**2 * np.cos(phi)**2 + b**2 * np.sin(phi)**2)
+    r = np.sqrt(r)
+
+    h = r + elev
+    alpha = latitude_to_geocentric(phi)
+    beta = theta
+    x = h * np.cos(alpha) * np.cos(beta)
+    y = h * np.cos(alpha) * np.sin(beta)
+    z = h * np.sin(alpha)
+    v = np.stack([x, y, z], axis=1)
+    return v
+
+
+def xyz_to_lle(point):
+    """
+    Converts a (x, y, z) tuple into a (lat, lon, elev) tuple.
+    The center of the earth is the origin in the xyz-space.
+    The output latitude is assumed to be a common latitude (geodetic).
+    """
+    # Equatorial Radius of the Earth (ellipsoid model) in meters
+    a = 6378137.0
+    # Polar Radius of the Earth (ellipsoid model) in meters
+    b = 6356752.0
```

Comment:
Let's define these once at the top of `tools.py` as `earth_radius_equatorial` and `earth_radius_polar`. You can set `a = ` and `b = ` within the functions if you want to keep the expressions short. Or `a_sqrd = `, `b_sqrd = ` in the functions as suggested elsewhere.